draft-ietf-tls-tls13.md

```diff
@@ -4375,6 +4375,91 @@ currently deployed and properly configured TLS implementations.
 %%### Ticket Establishment
 %%### Updating Keys
 
+# State Machine
+
+This section provides a summary of the legal state transitions for the
+client and server handshakes.  State names (in all capitals, e.g.,
+START) have no formal meaning but are provided for ease of
+comprehension.  Messages which are sent only sometimes are indicated
+in [].
+
+## Client
+
+~~~~
+                  +-----> START
+             Recv |        |
+            Hello |        | Send ClientHello
+            Retry |        v
+        +-> Request +---- WAIT_SH
+    Can |                    | Recv ServerHello
+   Send |                    V
+  Early |                 WAIT_EE
+   Data |                    | Recv EncryptedExtensions
+        |           +--------+--------+
+        |     Using |                 | Using Certificate
+        |       PSK |                 v
+        |           |            WAIT_CERT_CR
+        |           |        Recv |       | Recv CertificateRequest
+        |           | Certificate |       v
+        |           |             |    WAIT_CERT
+        |           |             |       | Recv Certificate
+        |           |             v       v
+        |           |              WAIT_CV
+        |           |                 | Recv CertificateVerify
+        |           +> WAIT_FINISHED <+
+        |                  | Recv Finished
+        |                  |
+        +->                | [Send EndOfEarlyData]
```

Comment:
This should end just before EndOfEarlyData, right?

What do you think about changing the ends to not be arrows? If you keep HRR on the left side, it looks kinda like a state transition. How about?

```
                            START <---+
                             |        |
            Send ClientHello |        | Recv HelloRetryRequest
        /                    v        |
        |                 WAIT_SH ----+
    Can |                    | Recv ServerHello
   Send |                    V
  Early |                 WAIT_EE
   Data |                    | Recv EncryptedExtensions
        |           +--------+--------+
        |     Using |                 | Using Certificate
        |       PSK |                 v
        |           |            WAIT_CERT_CR
        |           |        Recv |       | Recv CertificateRequest
        |           | Certificate |       v
        |           |             |    WAIT_CERT
        |           |             |       | Recv Certificate
        |           |             v       v
        |           |              WAIT_CV
        |           |                 | Recv CertificateVerify
        |           +> WAIT_FINISHED <+
        |                  | Recv Finished
        \                  |
                           | [Send EndOfEarlyData]
                           | [Send Certificate [+ CertificateVerify]]
                           | Send Finished
 Can Send                  v
 App Data -->          CONNECTED
 After
 Here
```